src/sage/rings/finite_rings/finite_field_base.pyx

```diff
@@ -1229,6 +1230,37 @@ cdef class FiniteField(Field):
             True
             sage: all(to_V(h(c) * e) == c * to_V(e) for e in E for c in F)
             True
+
+        TESTS::
+
+            The various calling conventions should not conflict with each other.
+
+            sage: K = GF(81)
+            sage: K.free_module(map=False)
+            Vector space of dimension 4 over Finite Field of size 3
+            sage: K.free_module(K.prime_subfield(), map=False)
+            Vector space of dimension 4 over Finite Field of size 3
+            sage: iota = Hom(K.prime_subfield(), K).an_element()
+            sage: K.free_module(iota, map=False)
+            Vector space of dimension 4 over Finite Field of size 3
+            sage: K.free_module(K.subfield(2, "g2"), map=False)
+            Vector space of dimension 2 over Finite Field in g2 of size 3^2
+
+            Same test, in reverse order with map=True
```

Comment:
```suggestion
        Same test, in reverse order with ``map=True``::
```

Reply:
Cancelled these changes after reverting the caching patch.